src/test/java/bio/terra/common/retry/SamRetryTest.java

```diff
@@ -0,0 +1,55 @@
+package bio.terra.common.retry;
+
+import bio.terra.common.exception.InternalServerErrorException;
+import com.google.api.client.http.HttpStatusCodes;
+import java.time.Duration;
+import org.broadinstitute.dsde.workbench.client.sam.ApiException;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.Tag;
+
+@Tag("unit")
+public class SamRetryTest {
+
+  private int count;
+
+  @Before
+  public void setup() {
+    count = 0;
+  }
+
+  @Test(expected = InternalServerErrorException.class)
+  public void testRetryTimeout() throws Exception {
+    SamRetry.retry(() -> testRetryFinishInner(100), Duration.ofSeconds(10));
```

Comment:
assertThrows is the preferred way to do exception assertions. https://howtodoinjava.com/junit5/expected-exception-example/
https://engdoc.corp.google.com/eng/doc/devguide/java/testing/exceptions.md?cl=head